src/receive_imf.rs

```diff
@@ -599,12 +599,17 @@ async fn add_parts(
 
         // In lookup_chat_by_reply() and create_or_lookup_group(), it can happen that the message is put into a chat
         // but the From-address is not a member of this chat.
-        if let Some(chat_id) = chat_id {
-            if !chat::is_contact_in_chat(context, chat_id, from_id).await? {
-                let chat = Chat::load_from_db(context, chat_id).await?;
+        if let Some(_chat_id) = chat_id {
+            if !chat::is_contact_in_chat(context, _chat_id, from_id).await? {
+                let chat = Chat::load_from_db(context, _chat_id).await?;
                 if chat.is_protected() {
-                    let s = stock_str::unknown_sender_for_chat(context).await;
-                    mime_parser.repl_msg_by_error(&s);
+                    if chat.typ == Chattype::Single {
+                        // Just assign the message to the 1:1 chat with the actual sender instead.
+                        chat_id = None;
```

Comment:
> Is there a test covering this case?

Yes, `verified_chats::test_reply`. It also has an explanation how this can happen:

```
/// If Bob answers unencrypted from another address with a classical MUA,
/// the message is under some circumstances still assigned to the original
/// chat (see lookup_chat_by_reply()); this is meant to make aliases
/// work nicely.
/// However, if the original chat is verified, the unencrypted message
/// must NOT be assigned to it (it would be replaced by an error
/// message in the verified chat, so, this would just be a usability issue,
/// not a security issue).
```

> I think this check should be outside of `chat.is_protected()`. If this happens in 1:1 chat when it is not protected, we probably don't want to do the same.

I'm not sure I understand this sentence correctly, but assuming that you propose to never assign a message to a 1:1 chat by reply:

We added this feature, esp. `lookup_chat_by_reply()`, a while ago to support email addresses that forward to multiple people. If one of these people answers from their own email address, it should be shown in the 1:1 chat with the support email address. There are tests for this, `test_alias_answer_from_dc`, and `test_alias_support_answer_from_nondc`.